[Bug] `spatialdata==0.3.0` pins `zarr<3` which is incompatible with latest `numcodecs` (0.16.0)

**Describe the bug**
Per https://github.com/zarr-developers/zarr-python/issues/2963.
- [We depend on spatialdata 0.3.0][sd dep], which [pins `zarr<3`].
- `zarr<3` [references `numcodecs.blosc` symbols][bad import] that [were removed] in numcodecs 0.16.0.

This resulted in `import zarr` raising `ImportError` [in unrelated GHA][GHA] on #3916:

```
  File "/home/runner/work/TileDB-SOMA/TileDB-SOMA/venv-soma/lib/python3.11/site-packages/zarr/convenience.py", line 8, in <module>
    from zarr._storage.store import data_root, meta_root, assert_zarr_v3_api_available
  File "/home/runner/work/TileDB-SOMA/TileDB-SOMA/venv-soma/lib/python3.11/site-packages/zarr/_storage/store.py", line 10, in <module>
    from zarr.meta import Metadata2, Metadata3
  File "/home/runner/work/TileDB-SOMA/TileDB-SOMA/venv-soma/lib/python3.11/site-packages/zarr/meta.py", line 10, in <module>
    from zarr.util import json_dumps, json_loads
  File "/home/runner/work/TileDB-SOMA/TileDB-SOMA/venv-soma/lib/python3.11/site-packages/zarr/util.py", line 34, in <module>
    from numcodecs.blosc import cbuffer_sizes, cbuffer_metainfo
ImportError: cannot import name 'cbuffer_sizes' from 'numcodecs.blosc' (/home/runner/work/TileDB-SOMA/TileDB-SOMA/venv-soma/lib/python3.11/site-packages/numcodecs/blosc.cpython-311-x86_64-linux-gnu.so)
Error: Process completed with exit code 1.
```

[sd dep]: https://github.com/single-cell-data/TileDB-SOMA/blob/37480ff86c6e6b439647764188ceb6cff326109c/apis/python/requirements_spatial.txt#L4
[pins `zarr<3`]: https://github.com/scverse/spatialdata/blob/v0.3.0/pyproject.toml#L49
[bad import]: https://github.com/zarr-developers/zarr-python/blob/v2.18.5/zarr/util.py#L34
[were removed]: https://github.com/zarr-developers/numcodecs/blob/v0.16.0/docs/release.rst#removals
[GHA]: https://github.com/single-cell-data/TileDB-SOMA/actions/runs/14312394956/job/40110373816?pr=3916#step:12:38

**To Reproduce**
See [GHA] example.

**Versions (please complete the following information):**
 - TileDB-SOMA version: `main`
 - Language and language version: Python 3.11 (but probably all versions)
 - OS: macOS, Ubuntu

--- a/apis/python/requirements_spatial.txt
+++ b/apis/python/requirements_spatial.txt
@@ -2,5 +2,7 @@ geopandas
 tifffile
 pillow
 spatialdata>=0.2.5
+# spatialdata 0.3.0 pins `zarr<3`, which is incompatible with numcodecs 0.16.0. See also: https://github.com/single-cell-data/TileDB-SOMA/issues/3917
+numcodecs<0.16
 xarray
 dask<=2024.11.2

--- a/apis/python/setup.py
+++ b/apis/python/setup.py
@@ -291,7 +291,6 @@ setuptools.setup(
         "Intended Audience :: Developers",
         "Intended Audience :: Information Technology",
         "Intended Audience :: Science/Research",
-        "License :: OSI Approved :: MIT License",
         "Topic :: Scientific/Engineering :: Bio-Informatics",
         "Operating System :: Unix",
         "Operating System :: POSIX :: Linux",

